### DISCUSSION TITLE
libstrongswan rsa test getting hang with strongswan 5.8.4 and 5.9.6

### DISCUSSION THREAD

**Discussion by thakur-virendra** [strongswan/strongswan#2099] [General]
Running suite 'rsa':
    Running case 'generate': +++
 [HANG]

qemu-aarch64 ./libstrongswan_tests
loaded plugins: aes des rc2 sha2 sha3 random nonce x509 revocation pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp curve25519 xcbc cmac hmac ctr ccm gcm curl
Running 39 'libstrongswan' test suites:
  Running suite 'bio_reader':
    Running case 'integer reads': +++++
    Running case 'integer reads from end': +++++
    Running case 'data reads and peek': ++
    Running case 'data length reads': ++++
    Running case 'constructors': ++
  Passed all 5 'bio_reader' test cases
  Running suite 'bio_writer':
    Running case 'integer writes': +++++
    Running case 'data writes/skip': ++
    Running case 'data length writes': ++++
    Running case 'wrap writes': ++++
    Running case 'get/extract': ++
  Passed all 5 'bio_writer' test cases
  Running suite 'chunk':
    Running case 'equals': ++
    Running case 'chunk_compare': +++++++++++
    Running case 'clear': +
    Running case 'chunk_length': +
    Running case 'chunk_create_cat': +
    Running case 'chunk_split': +
    Running case 'chunk_skip': ++
    Running case 'chunk_increment': ++++++++
    Running case 'chunk_copy_pad': ++++++++++++
    Running case 'chunk_printable': +++++++++++++++
    Running case 'baseXX': +++
    Running case 'chunk_mac': +
    Running case 'chunk_hash': +
    Running case 'chunk_hash_static': +
    Running case 'chunk_internet_checksum': +
    Running case 'chunk_map': +
    Running case 'chunk_from_fd': +++
    Running case 'printf_hook': ++++++++++++
  Passed all 18 'chunk' test cases
  Running suite 'enum':
    Running case 'enum_to_name': +++++++++++++++++++
    Running case 'enum_from_name': +++++++++++++++++++
    Running case 'enum_flags_to_string': ++++
    Running case 'enum_printf_hook': ++++++++++++++++++++++++++++++++++++++++
  Passed all 4 'enum' test cases
  Running suite 'enumerator':
    Running case 'tokens': +++++++++++
    Running case 'filtered': ++
    Running case 'nested': ++++
    Running case 'cleaner': +
    Running case 'single': ++
  Passed all 5 'enumerator' test cases
  Running suite 'linked list':
    Running case 'insert/get': ++
    Running case 'remove': ++++
    Running case 'find': +++
    Running case 'invoke': ++
    Running case 'clone': +
    Running case 'equals': ++
  Passed all 6 'linked list' test cases
  Running suite 'linked list and enumerators':
    Running case 'enumerate': +++
    Running case 'insert_before()': ++++++
    Running case 'modify': ++++
    Running case 'create_from_enumerator': +
  Passed all 4 'linked list and enumerators' test cases
  Running suite 'hashtable':
    Running case 'put/get': +
    Running case 'get_match': +
    Running case 'remove': ++
    Running case 'enumerator': +
    Running case 'remove_at': ++
  Passed all 5 'hashtable' test cases
  Running suite 'array':
    Running case 'add/get/remove ptr': +
    Running case 'add/get/remove obj': +
    Running case 'enumerate': +
    Running case 'sort': ++
    Running case 'bsearch': ++
    Running case 'invoke': +
    Running case 'invoke offset': +
    Running case 'insert create': ++
  Passed all 8 'array' test cases
  Running suite 'identification':
    Running case 'create': ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    Running case 'printf_hook': ++
    Running case 'equals': ++++++++++++++++
    Running case 'matches': ++++++++++++++++++++++++++++++++++
    Running case 'hash': ++++
    Running case 'part enumeration': +
    Running case 'wildcards': +
    Running case 'clone': +
  Passed all 8 'identification' test cases
  Running suite 'traffic selector':
    Running case 'create': ++++++++++++++++++++++
    Running case 'addresses': ++
    Running case 'to_subnet': +++++++++++++++++++
    Running case 'subset': +++++++++
    Running case 'includes': ++++++++++
    Running case 'is_contained_in': +++++++
    Running case 'is_host': ++++++++++++++++++
    Running case 'set_address': +++++++++
    Running case 'cmp': ++++++++++++++++++++
    Running case 'clone': +
    Running case 'hash': +
    Running case 'icmp': ++++++++++
    Running case 'printf hook': ++
  Passed all 13 'traffic selector' test cases
  Running suite 'threading':
    Running case 'recursive mutex': +
    Running case 'spinlock': +
    Running case 'condvar': ++++++
    Running case 'rwlock': ++
    Running case 'rwlock condvar': +++++
    Running case 'semaphore': ++++
    Running case 'thread joining': ++
    Running case 'thread detaching': ++
    Running case 'thread cancellation': +++
    Running case 'thread cancellation point': ++++++++++++++
    Running case 'thread cleanup': +++++
    Running case 'thread local storage': ++
  Passed all 12 'threading' test cases
  Running suite 'process':
    Running case 'return values': qemu-aarch64: /bin/sh: Invalid ELF image for this architecture
-qemu-aarch64: /bin/sh: Invalid ELF image for this architecture
+
      Failure in 'test_retval_true': retval != 0 (255 != 0) (../../../../strongswan-5.8.4/src/libstrongswan/tests/suites/test_process.c:42, i = 0)
    Running case 'not found': +
    Running case 'echo': qemu-aarch64: /bin/sh: Invalid ELF image for this architecture
--
      Failure in 'test_echo': read(out, buf, strlen(msg) + 1) != strlen(msg) (0 != 4) (../../../../strongswan-5.8.4/src/libstrongswan/tests/suites/test_process.c:107, i = 0)
      Failure in 'test_echo_err': read(err, buf, strlen(msg) + 1) != strlen(msg) (22 != 21) (../../../../strongswan-5.8.4/src/libstrongswan/tests/suites/test_process.c:140, i = 0)
    Running case 'env': qemu-aarch64: /bin/sh: Invalid ELF image for this architecture
-
      Failure in 'test_env': read(out, buf, sizeof(buf)) > 0 (../../../../strongswan-5.8.4/src/libstrongswan/tests/suites/test_process.c:173, i = 0)
    Running case 'shell': qemu-aarch64: /bin/sh: Invalid ELF image for this architecture
-
      Failure in 'test_shell': retval != 3 (255 != 3) (../../../../strongswan-5.8.4/src/libstrongswan/tests/suites/test_process.c:193, i = 0)
  Passed 1/5 'process' test cases
  Running suite 'watcher':
    Running case 'read': +
    Running case 'write': +
    Running case 'multiread': +
    Running case 'multiwrite': +
  Passed all 4 'watcher' test cases
  Running suite 'stream':
    Running case 'sync': +++
    Running case 'async': +++
    Running case 'all': +++
    Running case 'concurrency': +++
  Passed all 4 'stream' test cases
  Running suite 'utils':
    Running case 'objects': +
    Running case 'return functions': +
    Running case 'timeval_add_ms': +
    Running case 'timespan_from_string': +++++++++++++++++
    Running case 'htoun,untoh': ++
    Running case 'round': +
    Running case 'string helper': ++++++++++++++++++++++++++++
    Running case 'malloc_align': +
    Running case 'memxor': ++
    Running case 'memeq': ++++++++++++++++++++++++++
    Running case 'memstr': ++++++++++++
    Running case 'memwipe': +++
    Running case 'utils_memrchr': ++++++++++++
    Running case 'translate': ++++++++++++
    Running case 'strreplace': +++++++++++++++++++++++++++++++++++++++
    Running case 'path_dirname': ++++++++++++++++++++
    Running case 'path_basename': ++++++++++++++++++++
    Running case 'path_absolute': ++++++++++++++++++++
    Running case 'printf_hooks': ++++++++++++++++++++++
    Running case 'mark_from_string': +++++++++++++++++++++++++++++++++++++
    Running case 'if_id_from_string': +++++++++++++++++
    Running case 'allocate_unique_if_ids': +++++++++++
    Running case 'signature_schemes_for_key': ++++++++++
  Passed all 23 'utils' test cases
  Running suite 'settings':
    Running case 'get/set_str (basic behavior)': +++++
    Running case 'get/set_bool': ++
    Running case 'get/set_int': ++
    Running case 'settings_value_as_uint64': +
    Running case 'get/set_double': ++
    Running case 'get/set_time': ++
    Running case 'section enumerator': +
    Running case 'key/value enumerator': +
    Running case 'include/load_files[_section]': ++++++
    Running case 'load_string[_section]': +++
    Running case 'fallback': +++
    Running case 'references': +++++
    Running case 'strings': +
    Running case 'valid/invalid data': ++
    Running case 'crlf': +++
  Passed all 15 'settings' test cases
  Running suite 'vectors':
    Running case 'transforms': +++++++++
  Passed all 1 'vectors' test cases
  Running suite 'ecdsa':
    Running case 'generate': +++
    Running case 'load': +++
  Passed all 2 'ecdsa' test cases
  Running suite 'rsa':
    Running case 'generate': +++

**Comment by thakur-virendra** [comment]
How can we disable rsa test from configuration or is there any method to skip rsa test case ?

**Comment by tobiasbrunner** [comment]
These tests require entropy, in particular if key lengths are not reduced for the tests. Maybe try reading [the docs](https://docs.strongswan.org/docs/5.9/devs/devs.html#_unit_tests)?

**Comment by thakur-virendra** [comment]
@tobiasbrunner  which file we can set this environment

**Comment by thakur-virendra** [comment]
got it thanks
